sdv/model.py

```diff
@@ -841,11 +667,159 @@ async def get(self) -> TypedDataPointResult[List[str]]:
             raise
 
     async def set(self, value: List[str]):
+        await self._set(value, self.__class__.__name__)
+
+    def create_broker_data_point(self, value: List[str]):
+        array = StringArray(values=value)
+        return BrokerDatapoint(string_array=array)
+
+
+class BatchSetBuilder:
+    """Collects data points to be updated in a single 'atomic' set"""
+
+    def __init__(self, client):
+        self.__client = client
+        self.__nodes = {}
+
+    def preset(self, node: DataPoint, value) -> "BatchSetBuilder":
```

Comment:
Please add typed @overrides to allow intellisense to infer the value's type.

Reply:
Yes, we talked about that. I left it out because:
* I tried it, but it didn't help much (just limiting types to the overall set we allow in "preset"/"update", but not to the concrete type depending on the first argument).
* It has to be done twice: In Model and in BatchSetBuilder (what I didn't like much - especially for the low outcome)

Reply:
I think this is due to the current design needs to be done twice I agree to postpone this after refactoring out data models with the flaunt queries story.
 would be nice to add sub-task for AB#[11219]

Reply:
Actually you can have the value depend on the node type - thats exactly what we want. Even if its just for intellisense thats already plenty helpful for newcomers to the API and also for linters to highlight errors. Look at my suggestion above - that way they only have to be added in one place.

Reply:
Done